os/drivers/usbhost/uvc/usbhost_uvc_driver.c

```diff
@@ -0,0 +1,2377 @@
+static struct uvc_entity *uvc_alloc_entity(uint16_t type, uint8_t id, unsigned int num, unsigned int extra_size)
+{
+	struct uvc_entity *entity;
+	unsigned int num_inputs;
+	unsigned int size;
+
+	num_inputs = (type & UVC_TERM_OUTPUT) ? num : num - 1;
+	size = sizeof(*entity) + extra_size + num_inputs;
+	entity = kmm_zalloc(size);
+	if (entity == NULL) {
+		return NULL;
+	}
+
+	entity->id = id;
+	entity->type = type;
+	entity->bNrInPins = num_inputs;
+	entity->baSourceID = ((void *)(entity + 1)) + extra_size;
+
+	return entity;
+}
+
+/****************************************************************************
+ * Name: usbhost_alloc_buffers
+ *
+ * Description:
+ *   Allocate transfer buffer memory.
+ *
+ * Input Parameters:
+ *   priv - A reference to the class instance.
+ *
+ * Returned Value:
+ *   On sucess, zero (OK) is returned.  On failure, an negated errno value
+ *   is returned to indicate the nature of the failure.
+ *
+ ****************************************************************************/
+
+static int usbhost_alloc_buffers(FAR struct usbhost_uvc_state_s *priv)
+{
+	FAR struct usbhost_hubport_s *hport;
+	size_t maxlen;
+	int ret;
+
+	DEBUGASSERT(priv != NULL && priv->usbclass.hport != NULL);
+	hport = priv->usbclass.hport;
+
+	/* Allocate memory for control requests */
+
+	ret = DRVR_ALLOC(hport->drvr, (FAR uint8_t **) & priv->ctrlreq, &maxlen);
+	if (ret < 0) {
+		uvdbg("ERROR: DRVR_ALLOC of ctrlreq failed: %d\n", ret);
+		goto errout;
+	}
+
+	DEBUGASSERT(maxlen >= sizeof(struct usb_ctrlreq_s));
+
+	return OK;
+
+errout:
+	usbhost_free_buffers(priv);
+	return ret;
+}
+
+/****************************************************************************
+ * Name: usbhost_free_buffers
+ *
+ * Description:
+ *   Free transfer buffer memory.
+ *
+ * Input Parameters:
+ *   priv - A reference to the class instance.
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+static void usbhost_free_buffers(FAR struct usbhost_uvc_state_s *priv)
+{
+	FAR struct usbhost_hubport_s *hport;
+
+	DEBUGASSERT(priv != NULL && priv->usbclass.hport != NULL);
+	hport = priv->usbclass.hport;
+
+	if (priv->ctrlreq) {
+		(void)DRVR_FREE(hport->drvr, priv->ctrlreq);
+	}
+
+	priv->pktsize = 0;
+	priv->ctrlreq = NULL;
+	priv->inbuf = NULL;
+}
+
+/****************************************************************************
+ * Name: usbhost_uvc_allocdevno
+ *
+ * Description:
+ *   This function allocate a new device number which can uniquely identify
+ *   the current device. A tiny functions to coordinate management of device
+ *   names.
+ *
+ * Input Parameters:
+ *   priv - Pointer to driver object.
+ *
+ * Returned Values:
+ *   None
+ *
+ ****************************************************************************/
+static int usbhost_uvc_allocdevno(FAR struct usbhost_uvc_state_s *priv)
+{
+	irqstate_t flags;
+	int devno;
+
+	flags = irqsave();
+	for (devno = 0; devno < 26; devno++) {
```

Comment:
minor number is 0~26, but why 26?

Reply:
I took this reference from existing reference of ./drivers/usbhost/usbhost_storage.c , ./drivers/usbhost/usbhost_storage.c.

Reply:
Then it should use CONFIG_USBHOST_NPREALLOC instead of 26, am I right?

Reply:
No, CONFIG_USBHOST_NPREALLOC is number of pre allocated instance. Here this is for device node name  eg /dev/video1 eg /dev/sda1 etc.

Reply:
ok but like a CONFIG_USBHOST_NPREALLOC, we should some code to check overflow regarding devno. How do you think?

Reply:
I feel there is not need to have an additional config for this, USB are plug and play device this number is just to avoid creating device name more than some number. Some times due to some issues, the number wont be release by driver and next available number can used instead for next plugin. So we set max as 26, this is followed in all other USB drivers in TizenRT currently. I'm not sure really we need to have a config for this?

Reply:
Acceptable, let's keep this.